packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/component_builder.rs

```diff
@@ -242,6 +229,58 @@ impl ComponentBuilder {
         }
     }
 
+    /// If component `component_idx` extended a prop using the `extend` attribute,
+    /// then add a child to the component corresponding to that prop.
+    ///
+    /// For example, if `<text name="$i"/>`, then we add a child corresponding to `$i.value`
+    /// in the case of `<p extend="$i.value" />` but not to the text component coming from a `$i.value`
+    /// that occurs outside the `extend` attribute.
+    fn add_child_if_prop_from_extend_attribute(&mut self, component_idx: ComponentIdx) {
```

Comment:
I don't like the `mut` here. I would much rather this function compute something and return an `Option<child>` and let the caller do the insertion.

It could be called `get_implicit_child` or something like that.

Reply:
I couldn't get myself to make the name that short, but I tried.